### DISCUSSION TITLE
#all problems

### DISCUSSION THREAD

**Discussion by dpille** [scandum/tintin#51] [Script help]
So I play two sessions and rarely switch between sessions.  I'm having some unexplained issue with some aliases involving #all, and at this point I'm totally at a loss.  I'm sure it's some odd syntax problem that I'm not experienced enough to find, and I'm hoping someone can help.

Lines from sessionA.tin:
#variable {pet} {lemangelo}
#var {combatspellpriority} {1}
#var {combatspell[1]} {cast 'shocking blast'}
#alias {target} {#all {#variable {target} {%1}}}
#alias {WWW} {order $pet kill $target}
#alias {qqq} {target %1; #all WWW; #all QQ}
#alias {QQ} {$combatspell[$combatspellpriority] $target}

Lines from sessionB.tin:
#variable {pet} {orangelo}
#var {combatspellpriority} {1}
#var {combatspell[1]} {cast 'shocking blast'}
#alias {target} {#all {#variable {target} {%1}}}
#alias {WWW} {order $pet kill $target}
#alias {qqq} {target %1; #all WWW; #all QQ}
#alias {QQ} {$combatspell[$combatspellpriority] $target}

Lines from sessionC.tin:
#variable {pet} {corpse}
#alias {target} {#all {#variable {target} {%1}}}
#alias {WWW} {order $pet kill $target}
#alias {qqq} {target %1; #all WWW; #all QQ}
#alias {QQ} {#NOP}

So the idea is qqq, from any active session, will tell each session to order its pet to kill and then cast a spell if it has one.

Symptoms:

If I start sessionC, then sessionA or sessionB, everything works as expected.

If I start sessionA or sessionB, then any other session, calling qqq fires QQ only in the first session, and directly issuing #all QQ from any session fires only in the first session.  However, at the same time, qqq fires WWW in every session under every circumstance I've tested, and calling qqq %1 always changes the value of $target via the target alias in all sessions.  Calling QQ directly in any session also fires under every circumstance I've tested.  If I change the "#all QQ" line in qqq to "QQ; #non-firing-session QQ" for the firing session and "QQ; #firing-session QQ" for the non-firing session, it all works.  Why doesn't #all QQ work??

I've checked the actually-running values of all these aliases and variables, so there's no odd line from elsewhere in any of the files causing a problem.

I'm totally stumped and it's driving me nuts. Any thoughts would be appreciated.

**Comment by scandum** [comment]
Which version are you running? I made some changes to #all recently if I recall correctly.

I'll try to look into this when I have time.

**Reply by dpille** [reply]
Originally noticed on 2.02.05, tried on 2.02.11 with similar results.  I believe at some point it worked on an earlier version though I'd be hard pressed to identify which.

**Reply by scandum** [reply]
I updated the #all code in the beta, not sure if it'll fix your problem but it just might. Feel free to give it a whirl.

https://mudhalla.net/tintin-beta.tar.gz

**Comment by Heaven1** [comment]
I had the EXACT same problem. I was pulling my hair out for a while trying to figure it out. But i did it!
Your "#alias {target} {#all {#variable {target} {%1}}} " statement should be changed to "#alias {target %1} {#all {#variable {target} {%1}}}"
For some reason including a wildcard variable in the alias name has different functionality than without.
Let me know if it works for you.
